vignettes/biocDockstore.Rmd

```diff
@@ -0,0 +1,119 @@
+was published in 2015.
+
+Bioconductor implemented a ["common workflow"](https://www.bioconductor.org/packages/release/BiocViews.html#___Workflow) concept a number
+of years ago.  (The term "common" is used on the Bioconductor
+landing page, to indicate that many bioinformaticians would
+be expected to engage with tasks reviewed in these workflows.
+The "common workflow" phrase is not intended to indicate
+a relationship to CWL, the "common workflow language" standard.)
+
+In Bioconductor parlance, a "common workflow" is an R package
+accompanied by a markdown or Sweave vignette that narrates
+and executes the workflow tasks using R.  Any sufficiently
+endowed deployment of R on a sufficient computing platform
+will run the workflow tasks to completion; this assertion is
+checked on a regular basis through Bioconductor's continuous
+integration system.  When the assertion fails, the workflow
+maintainer is alerted and corrections are made.
+
+Advantages to marrying the Bioconductor workflow concept with
+Dockstore VM/Docker/workflow infrastructure include
+
+- reduction in user burden of configuring and maintaining the
+execution platform
+- utilization of parameterized formal workflow specification in CWL,
+WDL, or Nextflow
+- general advantages to users of fostering participation in GA4GH
+best practices related to reproducibility and transparency
+
+Open questions concern the balance between specification of
+workflow steps in R and in the formal workflow language.  Bioconductor
+workflows can be written to take advantage of R's capabilities
+to drive computations on potentially heterogeneous
+clusters with programmable fault tolerance and job control.
+The particular advantages of CWL/WDL/Nextflow and
+other aspects of the Dockstore ecosystem need to be experienced,
+measured, and documented to help developers establish the
+appropriate balance between programming R and programming
+an exogenous workflow environment.
+
+# Working with the Dockstore API in Bioconductor
+
+The [AnVIL](https://github.com/Bioconductor/AnVIL) package
+handles basic aspects of authentication and API element
+cataloguing for the AnVIL project.
+
+```{r lka}
+library(AnVIL)
+```
+
+An object called 'dockstore' is defined and will be
+used to process API requests.  We'll define a little
+bit of temporary infrastructure to show what 'dockstore'
+can do.
+
+```{r getmeths}
+api = function(x) slot(x, "api")
+sapply(api(dockstore)$tags, "[[", "name")
+```
+
+These are groups of API components.  We're interested in
+the 'users' component.  Higher level methods will be
+introduced to help here, but for now we stick to base R
+methods.
+
+```{r lku}
+umeth = sort(grep("[uU]ser", names(operations(dockstore)), value=TRUE))
+decode = function(x) rjson::fromJSON(httr::content(x, "text"))
+umeth
+```
+
+We can use the following to determine our user identifier.
+```{r lklk}
+myuid = dockstore$getUser()
+decode(myuid)$id
```

Comment:
Maybe there should be an as.list.response method
```
as.list.response <-
    function(x, ...)
{
    as.list(content(x))
}
```
then `(dockstore$getUser() %>% as.list())$id`

Reply:
done